## ADIDAS ANALISIS
**By: Diana Bohorquez**

In [ ]:
!wget https://github.com/prayagpadwal/Adidas-Sales-Analysis/raw/main/Adidas%20US%20Sales%20Datasets.xlsx

--2025-09-04 01:13:42--  https://github.com/prayagpadwal/Adidas-Sales-Analysis/raw/main/Adidas%20US%20Sales%20Datasets.xlsx
Resolving github.com (github.com)... 140.82.116.4
Connecting to github.com (github.com)|140.82.116.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/prayagpadwal/Adidas-Sales-Analysis/main/Adidas%20US%20Sales%20Datasets.xlsx [following]
--2025-09-04 01:13:42--  https://raw.githubusercontent.com/prayagpadwal/Adidas-Sales-Analysis/main/Adidas%20US%20Sales%20Datasets.xlsx
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 693038 (677K) [application/octet-stream]
Saving to: ‘Adidas US Sales Datasets.xlsx’

Adidas US Sales Dat 100%[===================>] 676.79K  --.-KB/s    in 0.0

In [ ]:
import pandas as pd
df = pd.read_excel("Adidas US Sales Datasets.xlsx")

In [ ]:
df.head(10)

Retailer  Retailer ID Invoice Date     Region     State      City  \
0  Foot Locker      1185732   2020-01-01  Northeast  New York  New York   
1  Foot Locker      1185732   2020-01-02  Northeast  New York  New York   
2  Foot Locker      1185732   2020-01-03  Northeast  New York  New York   
3  Foot Locker      1185732   2020-01-04  Northeast  New York  New York   
4  Foot Locker      1185732   2020-01-05  Northeast  New York  New York   
5  Foot Locker      1185732   2020-01-06  Northeast  New York  New York   
6  Foot Locker      1185732   2020-01-07  Northeast  New York  New York   
7  Foot Locker      1185732   2020-01-08  Northeast  New York  New York   
8  Foot Locker      1185732   2020-01-21  Northeast  New York  New York   
9  Foot Locker      1185732   2020-01-22  Northeast  New York  New York   

                     Product  Price per Unit  Units Sold  Total Sales  \
0      Men's Street Footwear            50.0        1200     600000.0   
1    Men's Athletic Footwear            50.0        1000     500000.0   
2    Women's Street Footwear            40.0        1000     400000.0   
3  Women's Athletic Footwear            45.0         850     382500.0   
4              Men's Apparel            60.0         900     540000.0   
5            Women's Apparel            50.0        1000     500000.0   
6      Men's Street Footwear            50.0        1250     625000.0   
7    Men's Athletic Footwear            50.0         900     450000.0   
8    Women's Street Footwear            40.0         950     380000.0   
9  Women's Athletic Footwear            45.0         825     371250.0   

   Operating Profit  Operating Margin Sales Method  
0          300000.0              0.50     In-store  
1          150000.0              0.30     In-store  
2          140000.0              0.35     In-store  
3          133875.0              0.35     In-store  
4          162000.0              0.30     In-store  
5          125000.0              0.25     In-store  
6          312500.0              0.50     In-store  
7          135000.0              0.30       Outlet  
8          133000.0              0.35       Outlet  
9          129937.5              0.35       Outlet

0. Cuantas filas y columnas tiene la tabla?

In [ ]:
print("Dimensiones de la tabla:", df.shape)

Dimensiones de la tabla: (9648, 13)


0.1 Cual es la unidad de observación de la tabla?

In [ ]:
print("\nColumnas disponibles:")
for col in df.columns:
    print("-", col)


Columnas disponibles:
- Retailer
- Retailer ID
- Invoice Date
- Region
- State
- City
- Product
- Price per Unit
- Units Sold
- Total Sales
- Operating Profit
- Operating Margin
- Sales Method


La unidad de observación de la tabla es **cada transacción de venta de un producto de Adidas**.  
Cada fila corresponde a un registro de venta realizado por un minorista (retailer), en una fecha y lugar específicos, con información sobre el producto vendido, el precio, la cantidad de unidades, el método de venta y las ganancias asociadas.

1. Cuales fueron las ventas totales?

In [ ]:
ventas_totales = df["Total Sales"].sum()
print("Las ventas totales fueron: $", round(ventas_totales, 2))

Las ventas totales fueron: $ 899902125.0


1.1 Cuanto fue la ganancia operacional total?

In [ ]:
ganancia_operacional_total = df["Operating Profit"].sum()
print("La ganancia operacional total fue: $", round(ganancia_operacional_total, 2))

La ganancia operacional total fue: $ 332134761.45


1.2 Cuanto fue el margen promedio total ?

In [ ]:
margen_promedio = df["Operating Margin"].mean() * 100
print("El margen operativo promedio fue:", round(margen_promedio, 2), "%")

El margen operativo promedio fue: 42.3 %


**1.3 Por que el margen promedio y no el total. Explique con palabras**



El **margen operativo** es un indicador expresado en forma de **proporción o porcentaje**.  
A diferencia de las ventas o de la ganancia, que son montos de dinero y por eso sí se pueden sumar, el margen muestra la **relación entre la ganancia y las ventas** de cada transacción.  

Si sumáramos todos los márgenes fila por fila, el resultado no tendría sentido, porque estaríamos acumulando porcentajes de distintas operaciones.  
Por eso, lo correcto es calcular el **promedio del margen**, que nos da un valor representativo de la eficiencia general de la empresa.  

En este caso, el margen promedio fue de aproximadamente **42%**, lo que significa que, en promedio, de cada dólar vendido, 42 centavos se convierten en ganancia operativa.



2. Cuales fueron las ventas totales por Retailer?

In [ ]:
ventas_por_retailer = df.groupby("Retailer")["Total Sales"].sum().sort_values(ascending=False)
print("Ventas totales por Retailer:")
print(ventas_por_retailer)

Ventas totales por Retailer:
Retailer
West Gear        242964333.0
Foot Locker      220094720.0
Sports Direct    182470997.0
Kohl's           102114753.0
Amazon            77698912.0
Walmart           74558410.0
Name: Total Sales, dtype: float64


2.1 Cual fue la ganancia operacional total por retailer?

In [ ]:
ganancia_por_retailer = df.groupby("Retailer")[["Operating Profit"]].sum()
ganancia_por_retailer = ganancia_por_retailer.sort_values(by="Operating Profit", ascending=False)
print("Ganancia operacional total por Retailer:")
print(ganancia_por_retailer)

Ganancia operacional total por Retailer:
               Operating Profit
Retailer                       
West Gear           85667873.18
Foot Locker         80722124.81
Sports Direct       74332954.96
Kohl's              36811252.58
Amazon              28818503.31
Walmart             25782052.61


2.3 Tomando los dos resultados de arriba: El ganancia operacional total y las ventas totales divida ambos resultados. Le tiene que dar el margen total de cada Retailer.

In [ ]:
ventas_totales = df.groupby("Retailer")["Total Sales"].sum()
ganancia_total = df.groupby("Retailer")["Operating Profit"].sum()
margen_total = ganancia_total / ventas_totales
margen_total = margen_total * 100
print("📊 Margen Operacional Total por Retailer (%):")
print(margen_total.round(2))

📊 Margen Operacional Total por Retailer (%):
Retailer
Amazon           37.09
Foot Locker      36.68
Kohl's           36.05
Sports Direct    40.74
Walmart          34.58
West Gear        35.26
dtype: float64


2.4 Cual fue el margen operacional promedio por retailer?

In [ ]:
margen_promedio = df.groupby("Retailer")["Operating Margin"].mean() * 100
print("📊 Margen Operacional Promedio por Retailer (%):")
print(margen_promedio.round(2))

📊 Margen Operacional Promedio por Retailer (%):
Retailer
Amazon           41.79
Foot Locker      41.79
Kohl's           41.93
Sports Direct    44.49
Walmart          40.65
West Gear        41.79
Name: Operating Margin, dtype: float64


2.5 Por que los dos resultados anteriores le dieron diferente? Argumente

La diferencia se debe a la forma en que se calcula cada indicador:

- **Margen operacional total**: se obtiene dividiendo la **ganancia operacional total** entre las **ventas totales**. Es una medida *ponderada*, ya que las observaciones con mayores ventas influyen más en el resultado. Responde a la pregunta: *“¿Qué porcentaje de todas las ventas se convirtió en ganancia operacional?”*

- **Margen operacional promedio**: se calcula como el promedio de los márgenes individuales de cada observación. Aquí todas las filas tienen el mismo peso, sin importar el tamaño de la venta. Responde a la pregunta: *“En promedio, ¿qué porcentaje de cada transacción representa la ganancia operacional?”*

En conclusión, el margen total refleja la eficiencia global del retailer considerando el volumen de ventas, mientras que el margen promedio refleja la eficiencia promedio de cada operación individual. Por ello, aunque ambos hablan de rentabilidad, los valores no necesariamente coinciden.

3.1 Retailer que mas unidades totales vende? Saque cuanto vende en unidades cada retailer y busque el mayor

In [ ]:
unidades_por_retailer = df.groupby("Retailer")["Units Sold"].sum()
print("📦 Unidades totales vendidas por Retailer:")
print(unidades_por_retailer)
max_retailer = unidades_por_retailer.idxmax()
max_unidades = unidades_por_retailer.max()
print(f"\n👉 El retailer que más unidades vendió fue **{max_retailer}** con {max_unidades:,} unidades.")

📦 Unidades totales vendidas por Retailer:
Retailer
Amazon           197990
Foot Locker      604369
Kohl's           287375
Sports Direct    557640
Walmart          206225
West Gear        625262
Name: Units Sold, dtype: int64

👉 El retailer que más unidades vendió fue **West Gear** con 625,262 unidades.


3.2 Cual fue el producto que mas unidades totales vendió?

In [ ]:
unidades_por_producto = df.groupby("Product")["Units Sold"].sum().sort_values(ascending=False)
print("📦 Unidades totales vendidas por Producto:")
print(unidades_por_producto)
max_producto = unidades_por_producto.idxmax()
max_unidades_prod = unidades_por_producto.max()

print(f"\n👉 El producto que más unidades vendió fue **{max_producto}** con {max_unidades_prod:,} unidades.")

📦 Unidades totales vendidas por Producto:
Product
Men's Street Footwear        593320
Men's Athletic Footwear      435526
Women's Apparel              433827
Women's Street Footwear      392269
Women's Athletic Footwear    317236
Men's Apparel                306683
Name: Units Sold, dtype: int64

👉 El producto que más unidades vendió fue **Men's Street Footwear** con 593,320 unidades.


3.3 Cual es el metodo de venta que mas ventas totales tuvo y más ventas promedio

In [ ]:
ventas_totales_metodo = df.groupby("Sales Method")["Total Sales"].sum().sort_values(ascending=False)
ventas_promedio_metodo = df.groupby("Sales Method")["Total Sales"].mean().sort_values(ascending=False)
print("💰 Ventas totales por método de venta:")
print(ventas_totales_metodo)
print("\n📊 Ventas promedio por método de venta:")
print(ventas_promedio_metodo)

max_total_metodo = ventas_totales_metodo.idxmax()
max_total_valor = ventas_totales_metodo.max()

max_promedio_metodo = ventas_promedio_metodo.idxmax()
max_promedio_valor = ventas_promedio_metodo.max()

💰 Ventas totales por método de venta:
Sales Method
In-store    356643750.0
Outlet      295585493.0
Online      247672882.0
Name: Total Sales, dtype: float64

📊 Ventas promedio por método de venta:
Sales Method
In-store    204967.672414
Outlet       97908.411063
Online       50659.210882
Name: Total Sales, dtype: float64




👉 El método con más **ventas totales** fue **In-store** con 356,643,750.
👉 El método con más **ventas promedio** fue **In-store** con 204,968.



3.4 Cual Fue el estado que más ventas totales tuvo ?

In [ ]:

ventas_por_estado = df.groupby("State")["Total Sales"].sum()
ventas_ordenadas = ventas_por_estado.sort_values(ascending=False)

print("💰 Ventas totales por Estado (ordenadas):")
print(ventas_ordenadas)

💰 Ventas totales por Estado (ordenadas):
State
New York          64229039.0
California        60174133.0
Florida           59283714.0
Texas             46359746.0
South Carolina    29285637.0
Washington        26330718.0
North Carolina    23956531.0
Louisiana         23750781.0
Hawaii            22282457.0
Virginia          21575040.0
Oregon            21349674.0
Colorado          20996536.0
Nevada            20858509.0
New Mexico        19865016.0
Idaho             19276878.0
Georgia           18997466.0
Michigan          18625433.0
Wyoming           18577517.0
Ohio              18484583.0
Tennessee         18067440.0
Alabama           17633424.0
New Hampshire     16411667.0
Arizona           15782221.0
Montana           15710886.0
Mississippi       15591709.0
Alaska            14753103.0
Vermont           14352923.0
Arkansas          12639347.0
Delaware          12298412.0
Connecticut       11573448.0
Massachusetts     11076810.0
West Virginia     10689160.0
Oklahoma          1062073

4. Teniendo en cuenta los resultados de arriba: 1. Tome el Retailer que vendio
mas 2. El estado que vendio mas 3.el producto que vendio mas y el metodo de ventas que vendio más 4. Filtre y genere un dataframe con ellos 5. llamelo como un nuevo dataframe

In [ ]:
best_retailer = df.groupby("Retailer")["Total Sales"].sum().idxmax()
best_state = df.groupby("State")["Total Sales"].sum().idxmax()
best_product = df.groupby("Product")["Units Sold"].sum().idxmax()
best_method = df.groupby("Sales Method")["Total Sales"].sum().idxmax()

print("\n🏆 Resultados")
print("Mejor Retailer:", best_retailer)
print("Mejor Estado:", best_state)
print("Producto más vendido:", best_product)
print("Mejor método de venta:", best_method)

df_filtrado = df[
    (df["Retailer"] == best_retailer) &
    (df["State"] == best_state) &
    (df["Product"] == best_product) &
    (df["Sales Method"] == best_method)]

Best = df_filtrado

print("\n📊 DataFrame filtrado:")
display(Best)

print("Cantidad de filas encontradas:", len(Best))


🏆 Resultados
Mejor Retailer: West Gear
Mejor Estado: New York
Producto más vendido: Men's Street Footwear
Mejor método de venta: In-store

📊 DataFrame filtrado:


Retailer  Retailer ID Invoice Date     Region     State    City  \
1443  West Gear      1185732   2021-03-08  Northeast  New York  Albany   
1449  West Gear      1185732   2021-04-09  Northeast  New York  Albany   
1455  West Gear      1185732   2021-05-08  Northeast  New York  Albany   
1461  West Gear      1185732   2021-06-10  Northeast  New York  Albany   
1467  West Gear      1185732   2021-07-08  Northeast  New York  Albany   
1473  West Gear      1185732   2021-08-09  Northeast  New York  Albany   
1479  West Gear      1185732   2021-09-08  Northeast  New York  Albany   
1485  West Gear      1185732   2021-10-10  Northeast  New York  Albany   
1491  West Gear      1185732   2021-11-09  Northeast  New York  Albany   
1497  West Gear      1185732   2021-12-08  Northeast  New York  Albany   

                    Product  Price per Unit  Units Sold  Total Sales  \
1443  Men's Street Footwear            45.0         870     391500.0   
1449  Men's Street Footwear            45.0         825     371250.0   
1455  Men's Street Footwear            55.0         870     478500.0   
1461  Men's Street Footwear            60.0         875     525000.0   
1467  Men's Street Footwear            65.0         950     617500.0   
1473  Men's Street Footwear            65.0         925     601250.0   
1479  Men's Street Footwear            55.0         850     467500.0   
1485  Men's Street Footwear            60.0         800     480000.0   
1491  Men's Street Footwear            60.0         775     465000.0   
1497  Men's Street Footwear            65.0         875     568750.0   

      Operating Profit  Operating Margin Sales Method  
1443          176175.0              0.45     In-store  
1449          167062.5              0.45     In-store  
1455          215325.0              0.45     In-store  
1461          236250.0              0.45     In-store  
1467          277875.0              0.45     In-store  
1473          270562.5              0.45     In-store  
1479          210375.0              0.45     In-store  
1485          216000.0              0.45     In-store  
1491          209250.0              0.45     In-store  
1497          255937.5              0.45     In-store

Cantidad de filas encontradas: 10


In [ ]:
Best.head(2)

Retailer  Retailer ID Invoice Date     Region     State    City  \
1443  West Gear      1185732   2021-03-08  Northeast  New York  Albany   
1449  West Gear      1185732   2021-04-09  Northeast  New York  Albany   

                    Product  Price per Unit  Units Sold  Total Sales  \
1443  Men's Street Footwear            45.0         870     391500.0   
1449  Men's Street Footwear            45.0         825     371250.0   

      Operating Profit  Operating Margin Sales Method  
1443          176175.0              0.45     In-store  
1449          167062.5              0.45     In-store